compiler/rustc_codegen_llvm/src/va_arg.rs

```diff
@@ -175,6 +175,92 @@ fn emit_aapcs_va_arg<'ll, 'tcx>(
     val
 }
 
+fn emit_s390x_va_arg<'ll, 'tcx>(
+    bx: &mut Builder<'_, 'll, 'tcx>,
+    list: OperandRef<'tcx, &'ll Value>,
+    target_ty: Ty<'tcx>,
+) -> &'ll Value {
+    // Implementation of the s390x ELF ABI calling convention for va_args see
+    // https://github.com/IBM/s390x-abi (chapter 1.2.4)
+    let va_list_addr = list.immediate();
+    let va_list_layout = list.deref(bx.cx).layout;
+    let va_list_ty = va_list_layout.llvm_type(bx);
+    let layout = bx.cx.layout_of(target_ty);
+
+    let in_reg = bx.append_sibling_block("va_arg.in_reg");
+    let in_mem = bx.append_sibling_block("va_arg.in_mem");
+    let end = bx.append_sibling_block("va_arg.end");
+
+    // FIXME: vector ABI not yet supported.
+    let target_ty_size = bx.cx.size_of(target_ty).bytes();
+    let indirect: bool = target_ty_size > 8 || !target_ty_size.is_power_of_two();
+    let unpadded_size = if indirect { 8 } else { target_ty_size };
+    let padded_size = 8;
+
+    // FIXME: structs with a single element of floating-point type should be
+    // passed in a FPR instead of a GPR.
+    let gpr_type = indirect || target_ty.is_any_ptr() || target_ty.is_integral();
+    let (max_regs, reg_count, reg_save_index, reg_padding) = if gpr_type {
```

Comment:
nit: I like the way clang does this with a `reg_count_field` here (either 0 or 1) followed by something like:

```
let reg_count = bx.struct_gep(...,reg_count_field))
```

Reply:
Done.